ERCS/erc-7694.md

```diff
@@ -0,0 +1,110 @@
+---
+eip: 7694
+title: Solana storage handler for CCIP-Write
+description: Cross-chain write deferral protocol incorporating storage handler for Solana
+author: Avneet Singh (@sshmatrix), 0xc0de4c0ffee (@0xc0de4c0ffee)
+discussions-to: https://ethereum-magicians.org/t/erc-7694-solana-storage-handler-for-ccip-write/19706
+status: Draft
+type: Standards Track
+category: ERC
+created: 2024-04-18
+---
+
+## Abstract
+The following standard is an extension to the cross-chain write deferral protocol introducing storage handler for Solana.
+
+## Motivation
+[EIP-5559](./eip-5559) introduces two external handlers for deferring write operations to L2s and databases. This document extends that specification by introducing a third storage handler targeting Solana as the storage provider. 
+
+L2s and databases both have centralising catalysts in their stack. For L2s, this centralising agent is the shared security with Ethereum mainnet. In case of databases, the centralising agent is trivial; it is the physical server hosting the database. In light of this, a storage provider that relies on its own independent consensus mechanism is preferred. This specification instructs how the clients should treat write deferrals made to the Solana handler.
+
+Solana is a cheap L1 solution that is fairly popular among Ethereum community and is widely supported alongside Ethereum by almost all wallet providers. There are several chain-agnostic protocols on Ethereum which could benefit from direct access to Solana blockspace; ENS is one such example where it can serve users of Solana via its chain-agnostic properties while also using Solana's own native storage. This development will encourage more cross-chain functionalities between Ethereum and Solana at core. 
```

Comment:
```suggestion
Solana is a low cost L1 solution that is supported alongside Ethereum by multiple wallet providers. There are several chain-agnostic protocols on Ethereum which could benefit from direct access to Solana blockspace; ENS is one such example where it can serve users of Solana via its chain-agnostic properties while also using Solana's own native storage. This development will encourage more cross-chain functionalities between Ethereum and Solana at core. 
```

Suggest avoiding commentary on popularity with Ethereum community.